### PR TITLE
Use await instead of .result

### PR DESCRIPTION
**Pull request by HeadCookie**

### CODE REVIEW

File: Src/Penneo/(Model)/SignatureLine.cs [hunk 1]
```diff
@@ -121,10 +122,10 @@ public Signer GetSigner(PenneoConnector con)
...
 
-        public bool SetSigner(PenneoConnector con, Signer signer)
+        public async Task<bool> SetSigner(PenneoConnector con, Signer signer)
         {
             Signer = signer;
-            return LinkEntity(con, Signer);
+            return await LinkEntity(con, Signer);
         }
     }
 }
```
Comment (w3dev-dk) [inline Src/Penneo/(Model)/SignatureLine.cs:131]: Misses empty line at end of file

File: Src/PenneoTests/CaseFileTests.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Linq;
+using System.Threading.Tasks;
```
Comment (w3dev-dk) [inline Src/PenneoTests/CaseFileTests.cs:3]: Is this needed?

File: Src/PenneoTests/QueryTests.cs [hunk 3]
```diff
@@ -43,53 +51,49 @@ public void FindAllTest()
         {
             var con = TestUtil.CreatePenneoConnector();
             var q = new Query(con);
-            FindCollectionTest(con, q.FindAll<CaseFile>);
+            FindCollectionTest(con, async () => await q.FindAll<CaseFile>());
```
Comment (w3dev-dk) [inline Src/PenneoTests/QueryTests.cs:54]: Don't C3 have a auto formatter to solve issues like here where you sometimes has a space and sometimes not? `async ()` vs `async()`

**Review by w3dev-dk — COMMENTED** [review]
I would like  second reviewer on this one

File: Src/Penneo/(Model)/CaseFile.cs [hunk 4]
```diff
@@ -279,73 +280,73 @@ public CaseFileStatus GetStatus()
         }
 
         [Obsolete("Obsolete since 1.0.18. Use GetTemplates instead.")]
-        public IEnumerable<CaseFileTemplate> GetCaseFileTemplates(PenneoConnector con)
+        public async Task<IEnumerable<CaseFileTemplate>> GetCaseFileTemplates(PenneoConnector con)
         {
-            return GetLinkedEntities<CaseFileTemplate>(con, "casefile/casefiletypes").Objects;
+            return (await GetLinkedEntities<CaseFileTemplate>(con, "casefile/casefiletypes")).Objects;
         }
```
[the remark this answers is not in the document]
Comment (w3dev-dk) [reply Src/Penneo/(Model)/CaseFile.cs:286]: If anything else is marked as obsolete and has been it for a long time then maybe consider taking advantage of the new major version to cleanup

File: Src/PenneoTests/QueryTests.cs (hunk 3 above)
Comment (HeadCookie) [reply Src/PenneoTests/QueryTests.cs:54]: It's an async anonymous method thus the white space. :)

Comment (w3dev-dk) [reply Src/PenneoTests/QueryTests.cs:54]: My question was more that you are not consistent in the way you format the code. Some places you have the space, others you don't...

Comment (HeadCookie) [reply Src/PenneoTests/QueryTests.cs:54]: Ah fair - now I understand. I will fix that

File: Src/PenneoTests/CaseFileTests.cs (hunk 2 above)
Comment (HeadCookie) [reply Src/PenneoTests/CaseFileTests.cs:3]: Nope - good catch

File: Src/Penneo/(Model)/CaseFile.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (HeadCookie) [reply Src/Penneo/(Model)/CaseFile.cs]: Removed obselete methods

**Review by w3dev-dk — COMMENTED** [review]
I could approve it but I think we should discus it on the team. It's a big change and anyone using the SDK will have to refactor everything when upgrading...

My concern is that if we are going to ask our customers to re-factore everything then I would prefer to do that one time only - and should that one time be to refactor to use async or should it be fixing our OpenAPI spec so they can auto generate their own SDK?

File: Src/Penneo/(Model)/Document.cs [hunk 5]
```diff
@@ -135,30 +138,6 @@ public void SetDocumentType(DocumentType type)
             DocumentType = type;
         }
 
-        /// <summary>
-        /// Options as json
-        /// </summary>
-        [JsonProperty("Options")]
-        [Obsolete("Obsolete. Use Opts")]
-        public string OptionsJson { get; set; }
```
Comment (fabianbadoi) [inline Src/Penneo/(Model)/Document.cs:143]: Why are we dropping this?

File: Src/Penneo/(Model)/Document.cs [hunk 6]
```diff
@@ -135,30 +138,6 @@ public void SetDocumentType(DocumentType type)
...
-        [JsonProperty("Options")]
-        [Obsolete("Obsolete. Use Opts")]
-        public string OptionsJson { get; set; }
-
-        /// <summary>
-        /// Options for the document
-        /// </summary>
-        [JsonIgnore]
-        [Obsolete("Obsolete. Use Opts")]
-        public Dictionary<string, object> Options
```
Comment (fabianbadoi) [inline Src/Penneo/(Model)/Document.cs:150]: Why are we dropping this?

**Review by fabianbadoi — CHANGES_REQUESTED** [review]
Looks good, but I don't understand why we're removing document.options

File: Src/Penneo/(Model)/Document.cs (hunk 5 above)
Comment (HeadCookie) [reply Src/Penneo/(Model)/Document.cs:143]: They are marked as deprecated in the API and obsolete here (one should use opts instead). If we are making a major version we might as well do some housekeeping - at least that was the thought.

Comment (HeadCookie) [reply Src/Penneo/(Model)/Document.cs:143]: I'll add it to the changelog

**Review by fabianbadoi — APPROVED** [review]

**Review by w3dev-dk — APPROVED** [review]

**Pull request merged**